Server/Auth/ApiAuthorizationFilter.cs

```diff
@@ -1,72 +1,90 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using Immense.RemoteControl.Shared.Extensions;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Build.Framework;
+using Microsoft.Extensions.Logging;
 using Remotely.Server.Services;
+using Remotely.Shared;
 using System;
 using System.Net;
 using System.Text;
+using System.Threading.Tasks;
 
 namespace Remotely.Server.Auth;
 
-public class ApiAuthorizationFilter : IAuthorizationFilter
+public class ApiAuthorizationFilter : IAsyncAuthorizationFilter
 {
     private readonly IDataService _dataService;
+    private readonly ILogger<ApiAuthorizationFilter> _logger;
 
-    public ApiAuthorizationFilter(IDataService dataService)
+    public ApiAuthorizationFilter(
+        IDataService dataService, 
+        ILogger<ApiAuthorizationFilter> logger)
     {
         _dataService = dataService;
+        _logger = logger;
     }
 
-    public void OnAuthorization(AuthorizationFilterContext context)
+    public async Task OnAuthorizationAsync(AuthorizationFilterContext context)
     {
-
-        if (context.HttpContext.User.Identity.IsAuthenticated)
+        try
+        {
+            await Authorize(context);
+        }
+        catch (Exception ex)
         {
-            var orgID = _dataService.GetUserByNameWithOrg(context.HttpContext.User.Identity.Name)?.OrganizationID;
-            context.HttpContext.Request.Headers["OrganizationID"] = orgID;
-            return;
+            _logger.LogError(ex, "Error while authorizing API key.");
         }
+    }
 
-        if (context.HttpContext.Request.Headers.TryGetValue("Authorization", out var result))
+    private async Task Authorize(AuthorizationFilterContext context)
```

Comment:
This got refactored back to using API tokens.  It now uses a somewhat-standard `X-Api-Key` header.